depends/check-dependencies.sh

```diff
@@ -57,30 +57,40 @@ check_header    libelf          elf.h libelf.h libelf/libelf.h gelf.h libelf/gel
 check_header    libusb          usb.h
 check_header    ncurses         ncurses.h ncurses/ncurses.h
 check_header    zlib            zlib.h
+check_header    libarchive      archive.h
+check_header    libcurl         curl/curl.h
+check_header    libssl          openssl/ssl.h
 
 check_program   git
 check_program   svn
 check_program   wget
 check_program   patch
 check_program   tar
 check_program   unzip
+check_program   bzip2
 
 check_program   autoconf
 check_program   automake
 check_program   cmake
 check_program   make
 check_program   gcc
 check_program   g++
-
-# check_program   freetype-config
+check_program   m4
 
 check_program   bison
 check_program   flex
 check_program   tclsh
+check_program   cmp
```

Comment:
Do you happen to know where cmp is needed?

Reply:
I think it's used when building newlib, but I'm not 100% sure. I can test again later.

Reply:
GCC actually doesn't build without diffutils: https://gist.github.com/sharkwouter/5a0a29e90de541f3597f67677c1fd43d

I'll change it to diff, though, more well known and part of the same toolset.